tox.ini

```diff
@@ -103,6 +103,9 @@ deps =
   # -c{toxinidir}/docs/requirements.txt
 description = Build The Docs
 changedir = {toxinidir}/docs
+# Reinstall so setuptools-scm runs and updates the _version.py
+# intentionally not in alphabetical order due to execution order
+commands_pre = {envpython} -m pip install --no-deps -e {toxinidir}
 commands =
   # Retrieve possibly missing commits:
   -git fetch --unshallow
```

Comment:
With the same code in `conf.py`, this could be dropped given that this operation remains unconditional there.

Reply:
right!

Reply:
know if this will gets the tags too?
```
--unshallow
If the source repository is complete, convert a shallow repository to a complete one, removing all the limitations imposed by shallow repositories.

If the source repository is shallow, fetch as much as possible so that the current repository has the same history as the source repository.
```

Reply:
Not sure. I've been using the two commands for years so maybe it wasn't true at the time I first introduced it.